pkg/storage/unified/resource/index.go

```diff
@@ -44,44 +43,86 @@ func NewIndex(s *server, opts Opts, path string, tracer tracing.Tracer) *Index {
 		shards: make(map[string]Shard),
 		log:    log.New("unifiedstorage.search.index"),
 		tracer: tracer,
-		path:   path,
 	}
 
 	return idx
 }
 
-func (i *Index) IndexBatch(ctx context.Context, list *ListResponse) error {
-	ctx, span := i.tracer.Start(ctx, tracingPrexfixIndex+"CreateIndexBatches")
-	for _, obj := range list.Items {
-		indexableResource, err := NewIndexedResource(obj.Value)
+// IndexBatches goes through all the shards and indexes their batches if they are large enough
+func (i *Index) IndexBatches(ctx context.Context, maxSize int, tenants []string) error {
+	_, span := i.tracer.Start(ctx, tracingPrexfixIndex+"IndexBatches")
+	defer span.End()
+
+	group := errgroup.Group{}
+	group.SetLimit(500)
```

Comment:
i guess we need to store cfg on the index struct so we can set the configurable limit here

Reply:
ah, good catch! The config settings are on Index.Opts already, so updated this to use that 👍